packages/snaps-utils/src/localization.ts

```diff
@@ -0,0 +1,146 @@
+import type { Infer } from 'superstruct';
+import {
+  create,
+  object,
+  optional,
+  record,
+  string,
+  StructError,
+} from 'superstruct';
+
+import type { SnapManifest } from './manifest';
+import type { VirtualFile } from './virtual-file';
+
+export const LOCALIZABLE_FIELDS = ['description', 'proposedName'] as const;
+
+export const LocalizationFileStruct = object({
+  locale: string(),
+  messages: record(
+    string(),
+    object({
+      message: string(),
+      description: optional(string()),
+    }),
+  ),
+});
+
+export type LocalizationFile = Infer<typeof LocalizationFileStruct>;
+
+/**
+ * Validate a list of localization files.
+ *
+ * @param localizationFiles - The localization files to validate.
+ * @returns The validated localization files.
+ * @throws If any of the files are considered invalid.
+ */
+export function getValidatedLocalizationFiles(
+  localizationFiles: VirtualFile[],
+): VirtualFile<LocalizationFile>[] {
+  for (const file of localizationFiles) {
+    try {
+      file.result = create(JSON.parse(file.toString()), LocalizationFileStruct);
+    } catch (error) {
+      if (error instanceof StructError) {
+        throw new Error(
+          `Failed to validate localization file "${file.path}": ${error.message}.`,
+        );
+      }
+
+      if (error instanceof SyntaxError) {
+        throw new Error(
+          `Failed to parse localization file "${file.path}" as JSON.`,
+        );
+      }
+
+      throw error;
+    }
+  }
+
+  return localizationFiles as VirtualFile<LocalizationFile>[];
```

Comment:
CodeCov seems to still complain about this

Reply:
Only the error cases are tested right now 😅